### DISCUSSION TITLE
How to build a box?

### DISCUSSION THREAD

**Discussion by caolz** [deepmodeling/deepmd-kit#570] [Q&A] [answered]
I use clusters as a reference set and use dmol3 to calculate. There is no lattice information. I need to make a box by myself.
My idea is:
The coordinates are XYZ
x_min = min(X) -  8
x_max = max(X) + 8
y_min = min(Y) -  8
y_max = max(Y) + 8
z_min = min(Z) -  8
z_max = max(Z) + 8

lattice (x_max, y_min, z_min
            x_min, y_max, z_min
            x_min, y_min, z_max)

And in this way, the cluster structure is not in the middle of the lattice.

However, the program reports an error and needs to comply with the right-hand rule.

How can I solve this problem.

**Comment by njzjz** [comment, marked as answer]
DeePMD-kit supports training non-periodic structures. See #427 for details. We will add examples in the next major version.

**Reply by caolz** [reply]
Ok, when will it be updated?

**Comment by njzjz** [comment]
Also, this lattice is incorrect.

**Reply by caolz** [reply]
Is there any good way to build lattice?

**Reply by njzjz** [reply]
You can refer https://lammps.sandia.gov/doc/Howto_triclinic.html and (a b c) is what you want.
In your example, the correct one should be ((a,0,0), (0,b,0),(0,0,c)).

**Reply by caolz** [reply]
Thank you for your reply, I will refer to it.
